src/backend/joanie/tests/core/test_models_order_group.py

```diff
@@ -24,3 +29,246 @@ def test_model_order_group_can_edit(self):
             course=order_group.course_product_relation.course,
         )
         self.assertFalse(order_group.can_edit)
+
+    def test_model_order_group_check_start_before_end(self):
+        """
+        Starting period date of the OrderGroup can't be greater than the end date.
+        """
+        start = timezone.now()
+        end = start - timedelta(days=10)
+
+        with self.assertRaises(IntegrityError) as context:
+            factories.OrderGroupFactory(start=start, end=end)
+
+        self.assertTrue(
+            'new row for relation "core_ordergroup" violates'
+            ' check constraint "check_start_before_end"' in str(context.exception)
+        )
+
+    def test_model_order_group_set_start_and_end_date(self):
+        """
+        When we set a start date that is not greater than the end date, the order
+        group can be created. It's either both dates that are set, or neither.
+        """
+        mocked_now = datetime(2025, 1, 12, 8, 8, 7, 456789)
+        with mock.patch("django.utils.timezone.now", return_value=mocked_now):
+            start = timezone.now()
+            end = start + timedelta(days=1)
+            order_group_1 = factories.OrderGroupFactory(start=start, end=end)
+
+        self.assertEqual(order_group_1.start, start)
+        self.assertEqual(order_group_1.end, end)
+
+        order_group_2 = factories.OrderGroupFactory()
+
+        self.assertEqual(order_group_2.start, None)
+        self.assertEqual(order_group_2.end, None)
+
+    def test_models_order_group_is_enabled_when_is_active_is_equal_to_false(self):
```

Comment:
Please use more concise names :)
```suggestion
    def test_models_order_group_is_enabled_is_active_false(self):
```